bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -605,6 +607,80 @@ def _copy_from_previous(self, cell):
                 cell.qualifier = previous.qualifier
 
 
+class _ReadRowsRequestManager(object):
+    """ Update the ReadRowsRequest message in case of failures by
+        filtering the already read keys.
+
+    :type message: class:`data_messages_v2_pb2.ReadRowsRequest`
+    :param message: Original ReadRowsRequest containing all of the parameters
+                    of API call
+
+    :type last_scanned_key: bytes
+    :param last_scanned_key: last successfully scanned key
+
+    :type rows_read_so_far: int
+    :param rows_read_so_far: total no of rows successfully read so far.
+                            this will be used for updating rows_limit
+
+    """
+
+    def __init__(self, message, last_scanned_key, rows_read_so_far):
+        self.message = message
+        self.last_scanned_key = last_scanned_key
+        self.rows_read_so_far = rows_read_so_far
+        self.new_message = None
```

Comment:
`new_message` is never updated.  Please remove this variable.

Reply:
Done, thanks.